### ISSUE TITLE
addCondition to Join Object with databaseName.tableName.columnName not possible

### ISSUE THREAD

**Issue by chq81** [propelorm/Propel2#887]
When you add a Join to the Query, the following is done to determine the tableAlias and ColumnName:
```   
$dotpos = strrpos($left, '.');
$leftTableAlias = substr($left, 0, $dotpos);
$leftColumnName = substr($left, $dotpos + 1);
```
This is not done, when you add a condition to a Join.
```
if (strrpos($left, '.')) {
    list($this->leftTableName,  $this->left[]) = explode('.', $left);
}
...
if (strrpos($right, '.')) {
   list($this->rightTableName, $this->right[]) = explode('.', $right);
} 
```
So a join with **db.foo.bar (left)** and **db.foo.baz (right)** works in the first, but not in the second case.

**Comment by staabm** [comment]
@chq81 could you please provide a unit test.

I think a possible fix could be to use `explode('.', $left,2 );` and `explode('.', $right, 2);`

**Comment by chq81** [comment]
@staabm:
You mean, providing a unittest for the problem or the fix? Sorry for asking, I just recently started working on github.

For the solution:
With a limiter, I would get **db** as table name and **foo.bar** as column name, whereas **db.foo** is the table name and **bar** the column name.

**Comment by staabm** [comment]
> You mean, providing a unittest for the problem or the fix? Sorry for asking, I just recently started working on github.

Intially we need a unit test which "demonstrates" the error.
After we have a unit test we can start implementing a fix.

> With a limiter, I would get db as table name and foo.bar as column name, whereas db.foo is the table name and bar the column name.

I think you are right. So using `strrpos` and `substr` seems to be the way to go.

**Comment by marcj** [comment]
Unit-test is not necessary, a fiddle in http://sandbox.propelorm.org/ is enough, so we can reproduce it easily.

**Comment by chq81** [comment]
@marcj:
I created a fiddle here: http://sandbox.propelorm.org/66a286f
Hope that helps understanding the issue.
Thanks for the help!

**Comment by marcj** [comment]
Does it fix your issue if you use `addJoinCondition('Book', 'Book.Bla = ...', ..)` instead?

**Comment by kosmodisk** [comment]
No, not really: http://sandbox.propelorm.org/7928151. Having the same problem here.

Documentation could really use some proper examples about complex joins because it is giving me gray hair. API is inconsistent nor does it validate its input (e.g when input is invalid field, or expecting TYPE_PHPNAME but input is TYPE_COLNAME etc etc). 

For example:
a) altering existing join (current issue)
b) adding field=field clause to join
c) adding field=constant clause to join
d) adding new, unrelated join
e) changing that join

**Comment by dereuromark** [comment]
Is someone able to make a PR here with suggested changes?
